packages/flutter_tools/lib/src/build_runner/devfs_web.dart

```diff
@@ -938,3 +942,30 @@ class ReleaseAssetServer {
     return shelf.Response.notFound('');
   }
 }
+void injectVersionScriptTag(
```

Comment:
I don't think this is the way to do it. If you want to insert it into the html, it should be as markup. Otherwise a JSON asset file seems like a reasonable way to go

Reply:
in one comment ditman suggested to inject it in a script tag .so i tried to do that.

Reply:
if that's not okay , i will revert to the previous way .

Reply:
if you inject it into a script tag you will need to use a javascript API to read it. Whereas you can use package:http JSON parse to read the version info in a json asset